pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go

```diff
@@ -2,419 +2,668 @@ package pathologicaleventlibrary
+		allowed := m.Matches(i)
+		if allowed {
+			logrus.WithField("message", msg).WithField("locator", l).Infof("event interval matches %s", k)
+			return allowed, m
+		}
+	}
+	return false, nil
+}
+
+// AllowedByAny checks if the given event locator and message match any registered matcher, and if
+// the repeating event should be allowed.
+// Returns true if so, the matcher name, and the matcher itself.
+func (r *AllowedPathologicalEventRegistry) AllowedByAny(
+	i monitorapi.Interval,
+	topology v1.TopologyMode) (bool, EventMatcher) {
+	l := i.StructuredLocator
+	msg := i.StructuredMessage
+	for k, m := range r.matchers {
+		allowed := m.Allows(i, topology)
+		if allowed {
+			logrus.WithField("message", msg).WithField("locator", l).Infof("duplicated event allowed by %s", k)
+			return allowed, m
+		}
+	}
+	return false, nil
+}
+
+func (r *AllowedPathologicalEventRegistry) GetMatcherByName(name string) (EventMatcher, error) {
+
+	matcher, ok := r.matchers[name]
+	if !ok {
+		return nil, fmt.Errorf("no pathological event matcher registered with name: %s", name)
+	}
+	return matcher, nil
+}
+
+// AllowedPathologicalEvents is the list of all allowed duplicate events on all jobs. Upgrade has an additional
+// list which is combined with this one.
+func NewUniversalPathologicalEventMatchers(kubeConfig *rest.Config, finalIntervals monitorapi.Intervals) *AllowedPathologicalEventRegistry {
+	registry := &AllowedPathologicalEventRegistry{matchers: map[string]EventMatcher{}}
 
-var AllowedRepeatedEventPatterns = []*regexp.Regexp{
 	// [sig-apps] StatefulSet Basic StatefulSet functionality [StatefulSetBasic] should not deadlock when a pod's predecessor fails [Suite:openshift/conformance/parallel] [Suite:k8s]
 	// PauseNewPods intentionally causes readiness probe to fail.
-	regexp.MustCompile(`ns/e2e-statefulset-[0-9]+ pod/ss-[0-9] node/[a-z0-9.-]+ - reason/Unhealthy Readiness probe failed: `),
-
 	// [sig-apps] StatefulSet Basic StatefulSet functionality [StatefulSetBasic] should perform rolling updates and roll backs of template modifications [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
 	// breakPodHTTPProbe intentionally causes readiness probe to fail.
-	regexp.MustCompile(`ns/e2e-statefulset-[0-9]+ pod/ss2-[0-9] node/[a-z0-9.-]+ - reason/Unhealthy Readiness probe failed: HTTP probe failed with statuscode: 404`),
-
-	// [sig-node] Probing container ***
-	// these tests intentionally cause repeated probe failures to ensure good handling
-	regexp.MustCompile(`ns/e2e-container-probe-[0-9]+ .* probe failed: `),
-	regexp.MustCompile(`ns/e2e-container-probe-[0-9]+ .* probe warning: `),
+	registry.AddPathologicalEventMatcherOrDie(&SimplePathologicalEventMatcher{
+		name: "E2EStatefulSetReadinessProbeFailed",
+		locatorKeyRegexes: map[monitorapi.LocatorKey]*regexp.Regexp{
+			monitorapi.LocatorNamespaceKey: regexp.MustCompile(`e2e-statefulset-[0-9]+`),
+			monitorapi.LocatorPodKey:       regexp.MustCompile(`ss-[0-9]`),
```

Comment:
looking at line 62 above, should the pattern be `ss2-[0-9]`

Reply:
Hmm, perhaps, this came from line 58, but to be honest looking at this:

```golang
	registry.AddPathologicalEventMatcherOrDie(&SimplePathologicalEventMatcher{
		name:               "KubeletUnhealthyReadinessProbeFailed",
		messageReasonRegex: regexp.MustCompile(`^Unhealthy$`),
		messageHumanRegex:  regexp.MustCompile(`Readiness probe failed`),
	})
```

They kinda both feel pointless don't they. I guess I should remove this one too?

Reply:
Taking it out.